### PR TITLE
Refund Details: Product & Qty section

### PR DESCRIPTION
**Pull request by mindgraffiti**
Part 1 of #1681 

In this PR the Product & Qty section title displays in the Refund Details screen and shows the product cell, as long as the refund was created as a "Refund by item" type. Note that this PR doesn't include the "Subtotal" cell because the PR is already too big.

For reference, here is the full Refund Details screen:
Screen Shot 2020-01-02 at 2 47 17 PM

## To Test
1. On the website, choose an order and create a refund by item. Refund 2 items or more.
2. After creating the refund, go to the app and navigate to the Order Details for the chosen order.
3. The "Refunded" cell should be the new semantic color, "link", which is an active (pink) color. Tap on the "Refunded" cell.
4. The Refund Details screen should appear, with the "Refund #NNN" title and a Product & Qty section. The cell should show an image (if any), the Product Title (product's name), SKU (if any) and a break down of the refunded amount, such as `$total_price ($item_price x qty)`

### Screenshots
| Order Details | Refund Details |
| --- | --- |
|  |  |

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by kyleaparker — APPROVED** [review]

**Review by jaclync — APPROVED** [review]
LGTM!

Code: ✅ (some localizable string and nit suggestions)
Testing: ✅

**Review by pmusolino — APPROVED** [review]
Code ✅ 
Testing ✅ 

## Screenshots
| Order Refund             |  Refund detail |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 11 Pro - 2020-01-03 at 11 37 23 | Simulator Screen Shot - iPhone 11 Pro - 2020-01-03 at 11 38 56

File: WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        guard abs(item.quantity) > 1 else {
+            return currencyFormatter.formatAmount(positiveTotal, with: currency) ?? String()
+        }
+
+
+        let itemTotal = currencyFormatter.formatAmount(positiveTotal, with: currency) ?? String()
+        let itemSubtotal = currencyFormatter.formatAmount(item.price, with: currency) ?? String()
+
+        return itemTotal + " (" + itemSubtotal + " × " + quantity + ")"
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift]: I forgot to localize it too. Added ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /// Item's Tax
+    /// Return $0.00 if there is no tax.
+    ///
+    var tax: String? {
+        let prefix = NSLocalizedString("Tax:", comment: "Tax label for total taxes line")
+        let totalTax = currencyFormatter.formatAmount(item.totalTax, with: currency) ?? String()
+        return prefix + " " + totalTax
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift]: ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    /// Item's SKU
+    ///
+    var sku: String? {
+        guard let sku = item.sku, sku.isEmpty == false else {
+            return nil
+        }
+
+        let prefix = NSLocalizedString("SKU:", comment: "SKU label")
+        return prefix + " " + sku
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/OrderItemRefundViewModel.swift]: ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import Yosemite
+
+
+/// The main file for Refund Details data.
+/// Must conform to NSObject so it can be the UITableViewDataSource.
+///
+final class RefundDetailsDataSource: NSObject {
+    /// Refund
+    ///
+    private(set) var refund: Refund
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift]: ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+    /// Order
+    ///
+    private(set) var order: Order
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift]: ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+        cell.selectionStyle = .default
+        cell.configure(item: itemViewModel, imageService: imageService)
+    }
+}
+
+
+// MARK: - Lookup products
+//
+extension RefundDetailsDataSource {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift]: ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsResultControllers.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import Foundation
+import Yosemite
+
+
+/// Results controllers used to render the Refund Details view
+///
+final class RefundDetailsResultControllers {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsResultControllers.swift]: change class name to singular  ✅

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+/// All things view-related for Refunds.
+///
+final class RefundDetailsViewModel {
+    /// Refund
+    ///
+    private(set) var refund: Refund
+
+    /// Order
+    ///
+    private(set) var order: Order
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift]: ✅

[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift]: No, this is a private setter with a public getter. Refund and order properties are publicly referenced in the ViewController (which needs the refund ID and order ID, for starters).

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    ///
+    init(order: Order, refund: Refund) {
+        self.order = order
+        self.refund = refund
+    }
+
+    /// Section Titles
+    ///
+    let productLeftTitle = NSLocalizedString("PRODUCT", comment: "Product section title")
+    let productRightTitle = NSLocalizedString("QTY", comment: "Quantity abbreviation for section title")
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift]: They are in the next PR. I couldn't add all the section changes to this PR without going over 1k lines 🙃

File: WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+    /// Products from a Refund
+    ///
+    var products: [Product] {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift]: I generally leave properties and constants set at their default file access levels, unless there's a very good reason to hide them or lock them down.

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift [hunk 11]
```diff
@@ -4,35 +4,169 @@ import Yosemite
...
 // MARK: - RefundDetailsViewController: Displays the details for a given Refund.
 //
-class RefundDetailsViewController: UIViewController {
-
-    /// Refund
+final class RefundDetailsViewController: UIViewController {
+    /// Main TableView.
     ///
-    private let refund: Refund
+    @IBOutlet weak var tableView: UITableView!
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift]: ✅

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift [hunk 12]
```diff
@@ -4,35 +4,169 @@ import Yosemite
...
 
+    /// Refund to be rendered.
+    ///
+    var viewModel: RefundDetailsViewModel! {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift]: ✅

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift [hunk 13]
```diff
@@ -4,35 +4,169 @@ import Yosemite
...
+        configureTableView()
+        registerTableViewCells()
+        registerTableViewHeaderFooters()
+        configureViewModel()
+        reloadSections()
     }
 
+    /// Setup: Navigation.
+    ///
     func configureNavigation() {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift]: ✅

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift [hunk 14]
```diff
@@ -4,35 +4,169 @@ import Yosemite
...
         let refundTitle = NSLocalizedString("Refund #%@", comment: "It reads: Refund #<refund ID>")
-        title = String.localizedStringWithFormat(refundTitle, String(refund.refundID))
+        title = String.localizedStringWithFormat(refundTitle, String(viewModel.refund.refundID))
+    }
+
+    /// Setup: TableView.
+    ///
+    func configureTableView() {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift]: ✅

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift [hunk 15]
```diff
@@ -4,35 +4,169 @@ import Yosemite
...
+
+    /// Registers all of the available UITableViewCells.
+    ///
+    func registerTableViewCells() {
+        viewModel.registerTableViewCells(tableView)
+    }
+
+    /// Registers all of the available TableViewHeaderFooters.
+    ///
+    func registerTableViewHeaderFooters() {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Payment Section/Refunds/RefundDetailsViewController.swift]: ✅

**Pull request merged**